packages/edit-navigation/src/filters/add-menu-name-editor.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * WordPress dependencies
+ */
+/**
+ * Internal dependencies
+ */
+import NameEditor from '../components/name-editor';
+import { addFilter } from '@wordpress/hooks';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+const addMenuNameEditor = createHigherOrderComponent(
+	( BlockEdit ) => ( props ) => {
+		if ( props.name !== 'core/navigation' ) {
+			return <BlockEdit { ...props } />;
+		}
+		return <NameEditor BlockEdit={ BlockEdit } blockEditProps={ props } />;
```

Comment:
Feels like we should be able to avoid passing `BlockEdit` in by implementing this as:
```jsx
<>
    <NameEditor { ...props } />
    <BlockEdit { ...props } />
</>
```
Could probably be a bit more specific about the props for `NameEditor` than my example..